components/raven_tools/common/types.ts

```diff
@@ -0,0 +1,23 @@
+import { Pipedream } from "@pipedream/types";
+
+interface PdAxiosRequest {
+  $: Pipedream;
+}
+
+export interface RavenToolsRequestParams extends PdAxiosRequest {
+  params: object;
+}
+
+export type Domain = string;
+export type Keyword = string;
+export type RavenToolsEntity = Domain | Keyword;
+
+export type RavenToolsResponse =
+  | {
+      response: "success";
+    }
+  | undefined;
+
+export interface AddKeywordParams extends RavenToolsRequestParams {
+  params: { domain: Domain; keyword: Keyword };
```

Comment:
Could be - I usually prefer mapping the types for each entity the application uses, even if they are the same, but since they are literal strings in this case, the API can't change them without a major (breaking) update anyway.

Changing them to 'string' as per your suggestion

Reply:
Could be - I usually prefer mapping the types for each entity the application uses, even if they are the same, but since they are literal strings in this case, the API can't change them without a major (breaking) update anyway.

Changing them to 'string' as per your suggestion.